launch/launch/utilities/normalize_to_list_of_substitutions_impl.py

```diff
@@ -37,8 +38,8 @@ def normalize(x):
             "Failed to normalize given item of type '{}', when only "
             "'str' or 'launch.Substitution' were expected.".format(type(x)))
 
-    if isinstance(subs, str):
-        return [TextSubstitution(text=subs)]
+    if isinstance(subs, (str, Path)):
+        return [TextSubstitution(text=str(subs))]
     if is_a_subclass(subs, Substitution):
         return [cast(Substitution, subs)]
     return [normalize(y) for y in cast(Iterable, subs)]
```

Comment:
I believe this won't handle an `Iterable` that contains a `Path`. See #790

Reply:
Quite right, added test that catches that and then fixed.